components/src/preact/lineageFilter/lineage-filter.tsx

```diff
@@ -45,6 +45,9 @@ const LineageFilterInner: FunctionComponent<LineageFilterInnerProps> = ({
 
     const inputRef = useRef<HTMLInputElement>(null);
 
+    const [hasInput, setHasInput] = useState<boolean>(!!initialValue);
+    const [inputValue, setInputValue] = useState(initialValue || '');
```

Comment:
`hasInput` is redundant - it can be computed from `inputValue`
```suggestion
    const [inputValue, setInputValue] = useState(initialValue || '');
    const hasInput = inputValue.length > 0;
```